core/services/keystore/keys/suikey/key.go

```diff
@@ -0,0 +1,108 @@
+package suikey
+
+import (
+	"crypto"
+	"crypto/ed25519"
+	cryptorand "crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/blake2b"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/internal"
+)
+
+// Ed25519Scheme Ed25519 signature scheme flag
+// https://docs.sui.io/concepts/cryptography/transaction-auth/keys-addresses#address-format
+const Ed25519Scheme byte = 0x00
+
+// Key represents a Sui account
+type Key struct {
+	raw    internal.Raw
+	signFn func(io.Reader, []byte, crypto.SignerOpts) ([]byte, error)
+	pubKey ed25519.PublicKey
+}
+
+// KeyFor creates an Account from a raw key
+func KeyFor(raw internal.Raw) Key {
+	privKey := ed25519.NewKeyFromSeed(internal.Bytes(raw))
+	pubKey := privKey.Public().(ed25519.PublicKey)
+	return Key{
+		raw:    raw,
+		signFn: privKey.Sign,
+		pubKey: pubKey,
+	}
+}
+
+// New creates new Key
+func New() (Key, error) {
+	return newFrom(cryptorand.Reader)
+}
+
+// MustNewInsecure returns an Account if no error
+func MustNewInsecure(reader io.Reader) Key {
+	key, err := newFrom(reader)
+	if err != nil {
+		panic(err)
+	}
+	return key
+}
+
+// newFrom creates a new Account from a provided random reader
+func newFrom(reader io.Reader) (Key, error) {
```

Comment:
looking at the issue raised below, do we use this function in production or only in test?